compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1442,11 +1442,12 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 
     /// Allocates a read-only byte or string literal for `mir::interpret`.
+    /// Returns the same `AllocId` if called again with the same bytes.
```

Comment:
Maybe rename this to `allocate_bytes_dedup` just to be sure the caller is aware?